dbms/src/Functions/FunctionsHashing.h

```diff
@@ -707,6 +709,58 @@ class FunctionAnyHash : public IFunction
                 ErrorCodes::ILLEGAL_COLUMN);
     }
 
+    template <bool first>
+    void executeUUID(const IColumn * column, typename ColumnVector<ToType>::Container & vec_to)
+    {
+        if (const ColumnUInt128 * col_from = checkAndGetColumn<ColumnUInt128>(column))
+        {
+            const ColumnUInt128::Container & vec_from = col_from->getData();
+            size_t size = vec_from.size();
+            for (size_t i = 0; i < size; ++i)
+            {
+                std::string hash_string = toString(UUID(vec_from[i]));
```

Comment:
Creating temporary std::string object in a loop - not acceptable.

But it's not needed to calculate a hash on string representation of UUID. Better to apply directly on bytes of binary data.